frame/staking/src/tests.rs

```diff
@@ -1708,6 +1741,7 @@ fn rebond_emits_right_value_in_event() {
 #[test]
 fn reward_to_stake_works() {
 	ExtBuilder::default()
+		.check_events(true)
```

Comment:
What is the methodology you used to add `.check_events` to some tests and not to some others? I want to make sure that we are covering 

As noted earlier, I think we should create brand new tests for this part. These should focus on all the various state transitions that lead to a validator/nominator's state change (`validate`, `nominate`, `chill`), and all the transitions that cause one's ledger amount to change: `slash`, `reward`, `bond_extra`, `unbond`, `rebond`. 

The existing staking tests are IMO really bad and unorganized. By skimming over these, I see no guarantee that we have covered all the cases. Instead, if we start from scratch, at least we have a shot at having an actual test-coverage.

Moreover, given the new changes, we are closing https://github.com/paritytech/substrate/issues/12742 in this PR as well.

Reply:
Sounds good. I was looking for tests that cover all possible codepaths. My initial idea was to introduce these checks to every test in staking, but that turned out too verbose and sometimes really hard to follow due to the nature of some of the tests.